### PR TITLE
Dissolve delay input update

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Extend the Input component with a left button slot and expose input event.

> BREAKING CHANGES (slot rename)

# Changes

- add `start` slot
- rename slot `additional` to `end`
- dispatch `input` event

# Test

- left slot

### CODE REVIEW

File: src/lib/components/Input.svelte [hunk 1]
```diff
@@ -90,12 +93,14 @@
...
         // restore value (e.g. to fix invalid paste)
         restoreFromValidValue();
+        dispatch("input");
         return;
       }
 
       // reset to undefined ("" => undefined)
       if (currentValue.length === 0) {
         restoreFromValidValue(true);
+        dispatch("input");
```
Comment (peterpeterparker) [inline src/lib/components/Input.svelte:103]: same

File: src/routes/(split)/components/input/+page.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ Both slots are displayed `flex` with `space-between`.
 
     <Input placeholder="Input text" inputType="text" value="" showInfo>
         <svelte:fragment slot="label">A label</svelte:fragment>
-        <span slot="additional" class="label">More</span>
+        <span slot="right" class="label">More</span>
```
Comment (peterpeterparker) [inline src/routes/(split)/components/input/+page.md:54]: shouldn't be `slot="end"` now?

Comment (mstrasinskis) [reply src/routes/(split)/components/input/+page.md]: Good catch, updated.

File: src/lib/components/Input.svelte [hunk 3]
```diff
@@ -90,12 +93,14 @@
       if (isValidICPFormat(currentValue) === false) {
         // restore value (e.g. to fix invalid paste)
         restoreFromValidValue();
+        dispatch("input");
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply src/lib/components/Input.svelte]: The flow was updated - https://github.com/dfinity/gix-components/pull/150/files#diff-0470d4e406dc332cd12212120f011c54a27f7d59cba656edf47c3a29a8d34421R88

[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/lib/components/Input.svelte]: Thanks

**Review by peterpeterparker — APPROVED** [review]
Already approving it, if you can just update the doc before merge it's super

**Pull request merged**